deploy/helm/sumologic/templates/setup/setup-cleanup.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if .Values.sumologic.cleanUpSecrets }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: collection-sumologic-secret-cleanup
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-weight": "3"
+    "helm.sh/hook-delete-policy": hook-succeeded
+spec:
+  template:
+    spec:
+      serviceAccountName: {{ template "sumologic.metadata.name.setup.roles.serviceaccount" . }}
```

Comment:
I checked if setup-service-account is available and seems that it's removed after the setup job is done

```
Error creating: pods "collection-sumologic-secret-cleanup-" is forbidden: error looking up service account default/collection-sumologic-setup: serviceaccount "collection-sumologic-setup" not found
```

Reply:
Also the `helm delete` failed at all (I don't see way to remove collection now). We need to fix that and check what happen if the container fails itself for some reason.

Reply:
The service account is surprising.  This is a pre-delete hook that runs before any resources are deleted in k8s. at least that is how it should behave per the docs.

Reply:
I suppose they are deleted together with setup-job when it's done